File: src/main/java/com/skyspace222/service/FlightCabinService.java
package com.skyspace222.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace222.domain.FlightCabin;
import com.skyspace222.dto.FlightCabinDTO;
import com.skyspace222.dto.FlightCabinSearchDTO;
import com.skyspace222.dto.FlightCabinPageDTO;
import com.skyspace222.dto.FlightCabinConvertCriteriaDTO;
import com.skyspace222.service.GenericService;
import com.skyspace222.dto.common.RequestDTO;
import com.skyspace222.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface FlightCabinService extends GenericService<FlightCabin, Integer> {

	List<FlightCabin> findAll();

	ResultDTO addFlightCabin(FlightCabinDTO flightCabinDTO, RequestDTO requestDTO);

	ResultDTO updateFlightCabin(FlightCabinDTO flightCabinDTO, RequestDTO requestDTO);

    Page<FlightCabin> getAllFlightCabins(Pageable pageable);

    Page<FlightCabin> getAllFlightCabins(Specification<FlightCabin> spec, Pageable pageable);

	ResponseEntity<FlightCabinPageDTO> getFlightCabins(FlightCabinSearchDTO flightCabinSearchDTO);
	
	List<FlightCabinDTO> convertFlightCabinsToFlightCabinDTOs(List<FlightCabin> flightCabins, FlightCabinConvertCriteriaDTO convertCriteria);

	FlightCabinDTO getFlightCabinDTOById(Integer flightCabinId);







}

File: skyspace222_ui/src/services/FlightCabinService.js
import http from "../http-common"; 

class FlightCabinService {
  getAllFlightCabins(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/flightCabin/flightCabins`, searchDTO);
  }

  get(flightCabinId) {
    return this.getRequest(`/flightCabin/${flightCabinId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/flightCabin?field=${matchData}`, null);
  }

  addFlightCabin(data) {
    return http.post("/flightCabin/addFlightCabin", data);
  }

  update(data) {
  	return http.post("/flightCabin/updateFlightCabin", data);
  }
  
  uploadImage(data,flightCabinId) {
  	return http.postForm("/flightCabin/uploadImage/"+flightCabinId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new FlightCabinService();
